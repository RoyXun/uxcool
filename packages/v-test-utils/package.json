{
  "name": "@suning/v-test-utils",
  "version": "1.0.10",
  "main": "src/index.js",
  "module": "src/index.js",
  "license": "MIT",
  "scripts": {
    "start": "npm run server",
    "compile": "v-tools run compile",
    "server": "v-tools run server"
  },
  "peerDependencies": {
    "vue": ">= 2.5.*"
  },
  "devDependencies": {
    "@suning/v-tools": "^1.0.35",
    "@vue/test-utils": "^1.0.0-beta.16"
  },
  "dependencies": {
    "@suning/v-trigger": "^1.0.64",
    "@vue/server-test-utils": "^1.0.0-beta.30",
    "jquery": "^3.3.1",
    "vue-server-renderer": "^2.6.11"
  }
}
