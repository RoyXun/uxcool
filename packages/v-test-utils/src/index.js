import $ from 'jquery';

export * from '@vue/test-utils';

export { $, $ as jQuery };
// ms
export async function waitTime(delay) {
  await new Promise((resolve) => {
    setTimeout(() => {
      resolve();
    }, delay);
  });
}
