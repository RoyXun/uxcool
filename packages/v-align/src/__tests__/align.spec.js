import { mount, waitTime } from '@cloud-sn/v-test-utils';
import Align from '../index';

describe('Align', () => {
  const alignWrapperMounter = {
    props: {
      align: {
        type: Object,
        default: () => {},
      },
      disabled: {
        type: Boolean,
        default: false,
      },
      monitorWinResize: {
        type: Boolean,
        default: false,
      },
      monitorBufferTime: {
        type: Number,
        default: 50,
      },
    },
    methods: {
      target() {
        return this.$refs.container;
      },
    },
    render() {
      const {
        align, disabled, monitorWinResize, target, monitorBufferTime
      } = this;
      return (
        <div ref="container" style="width: 300px; height: 300px;">
          <Align
            ref="alignWrapper"
            style="position:absolute; width: 60px; height: 60px;"
            target={target}
            align={align}
            disabled={disabled}
            monitorWinResize={monitorWinResize}
            monitorBufferTime={monitorBufferTime}
          >
            <span class="child">test align</span>
          </Align>
        </div>
      );
    },
  };

  it('render', async () => {
    const wrapper = await mount(alignWrapperMounter);
    const alignWrapper = wrapper.find({ ref: 'alignWrapper' });
    expect(wrapper.html()).toMatchSnapshot();

    wrapper.vm.$nextTick(async () => {
      wrapper.setProps({
        align: {
          useCssTransform: true,
          points: ['tl', 'tl'],
        },
        disabled: false,
        monitorWinResize: true,
        monitorBufferTime: 100,
      });
      await waitTime();
      expect(wrapper.html()).toMatchSnapshot();
      alignWrapper.destroy();
    });
  });

  it('disabled is false and monitorWinResize is true when mounted', async () => {
    const wrapper = await mount(alignWrapperMounter, {
      propsData: {
        disabled: false,
        monitorWinResize: true,
      },
    });
    const alignWrapper = wrapper.find({ ref: 'alignWrapper' });
    expect(wrapper.html()).toMatchSnapshot();

    wrapper.vm.$nextTick(async () => {
      wrapper.setProps({
        align: {
          useCssTransform: true,
          points: ['tl', 'tr'],
        },
        disabled: false,
        monitorWinResize: false,
      });
      await waitTime();
      expect(wrapper.html()).toMatchSnapshot();
      alignWrapper.destroy();
    });
  });

  it('disabled is true', async () => {
    const wrapper = await mount(alignWrapperMounter, {
      propsData: {
        disabled: false,
        monitorWinResize: true,
      },
    });
    const alignWrapper = wrapper.find({ ref: 'alignWrapper' });
    expect(wrapper.html()).toMatchSnapshot();

    wrapper.vm.$nextTick(async () => {
      wrapper.setProps({
        disabled: true,
      });
      await waitTime();
      expect(wrapper.html()).toMatchSnapshot();
    });
    alignWrapper.destroy();
  });

  it('render no child', async () => {
    const wrapper = await mount({
      render() {
        return (
          <div ref="container" style="width: 300px; height: 300px;">
            <Align ref="alignWrapper" style="position:absolute; width: 60px; height: 60px;"></Align>
          </div>
        );
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
