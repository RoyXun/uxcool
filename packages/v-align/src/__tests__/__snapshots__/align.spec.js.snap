// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Align disabled is false and monitorWinResize is true when mounted 1`] = `<div style="width: 300px; height: 300px;"><span class="child" style="position: absolute; width: 60px; height: 60px;">test align</span></div>`;

exports[`Align disabled is true 1`] = `<div style="width: 300px; height: 300px;"><span class="child" style="position: absolute; width: 60px; height: 60px;">test align</span></div>`;

exports[`Align render 1`] = `<div style="width: 300px; height: 300px;"><span class="child" style="position: absolute; width: 60px; height: 60px;">test align</span></div>`;

exports[`Align render no child 1`] = `
<div style="width: 300px; height: 300px;">
  <!---->
</div>
`;
